src/main/kotlin/org/rust/cargo/runconfig/test/CargoTest.kt

```diff
@@ -0,0 +1,35 @@
+package org.rust.cargo.runconfig.test
+
+import com.intellij.execution.configurations.RunConfiguration
+import com.intellij.execution.configurations.RunConfigurationModule
+import com.intellij.openapi.options.SettingsEditor
+import com.intellij.openapi.project.Project
+import org.rust.cargo.runconfig.CargoRunConfigurationBase
+import org.rust.cargo.runconfig.ui.CargoTestConfigurationEditForm
+
+/**
+ * [RunConfiguration] implementation that runs a rust module
+ */
+class CargoTest(project: Project,
+                name: String,
+                configurationType: CargoTestType)
+: CargoRunConfigurationBase<CargoTestConfigurationModule>(name,
+                                                          CargoTestConfigurationModule(project),
+                                                          configurationType) {
+
+    override var command: String = "test"
+    override var arguments: String = ""
+
+    private var _testName: String = ""
+    var testName: String
+        get() = _testName
+        set(value) {
+            _testName = value
+            arguments = "--test " + testName
+        }
```

Comment:
I am not super fluent with Kotlin properties, but I think the same can be written as 

```
var testName: String = ""
    get() = field
    set(value) {
        field = value
        arguments = "--test " + testName
    }
```

Also, mutating `arguments` when you change `testName` seems a bit not obvious... Maybe it is possible to override `arguments` getter in such a way that it inspects `testName` and appends `--test` if necessary?